jsapp/js/components/processing/sidebar/sidebarDisplaySettings.tsx

```diff
@@ -61,6 +91,57 @@ export default function SidebarDisplaySettings() {
     );
   }
 
+  function isFieldChecked(questionName: string) {
+    return selectedFields.some((field) => field.name === questionName);
+  }
+
+  function getCheckboxes(disabled?: boolean) {
+    if (!props.assetContent?.survey) {
+      return [];
+    }
+
+    const checkboxes = getInitialFields().map((question) => {
+      return {
+        label: question.label,
+        checked: isFieldChecked(question.name),
+        name: question.name,
+        disabled: disabled,
+      };
+    });
+
+    return checkboxes;
+  }
+
+  // To make the code a little simpler later on, we need an inverse array here
+  // to send to the the display, and a normal array to keep track of the
+  // checkboxes in this modal.
+  function onCheckboxesChange(list: MultiCheckboxItem[]) {
+    const newList = list
+      .filter((question) => question.checked)
+      .map((question) => {
+        return {name: question.name, label: question.label};
+      });
+
+    const hiddenList = list
+      .filter((question) => !question.checked)
+      .map((question) => question.name);
+
+    setHiddenFields(hiddenList);
```

Comment:
After you add code for building this list inside `applyFieldsSelection` you can simply remove this and the `const` above